src/sage/modular/drinfeld_modform/tutorial.py

```diff
@@ -0,0 +1,228 @@
+
+    g_i(w) = \sum_{i = 0}^{\infty} a_n(g_i)u(w)^i
+
+where `u(w) := e(w)^{-1}` and `e` is the exponential of the Carlitz
+module `\rho:T\mapsto T + \tau`. The analytic parameter `u` is called
+the *parameter at infinity*.
+
+A *Drinfeld modular form* of rank `r`, weight `k`, type `m` for
+`\mathrm{GL}_r(A)` is a rigid analytic function
+`f:\Omega^r(\mathbb{C}_{\infty}) \rightarrow \mathbb{C}_{\infty}`
+such that
+
+* `f(\gamma(w)) = \mathrm{det}(\gamma)^m j(\gamma, w)^k f(w)` for all
+  `\gamma` in `\mathrm{GL}_r(A)` and
+  `w\in \Omega^r(\mathbb{C}_{\infty})`;
+
+* `f` is holomorphic at infinity.
+
+The second condition is similar to the classical case. In the rank two
+situation, this expansion is simply given by
+`f = \sum_{n\geq 0} a_n(f) u^n` where `a_n(f)\in \mathbb{C}_{\infty}`.
+
+The reader is refered to
+part I of [BRP2018]_ for more information about the analytic theory of
+Drinfeld modular form of arbitrary rank.
+
+.. RUBRIC:: Ring of Drinfeld modular forms
+
+Letting `M_k^{r, m}(\mathrm{GL}_r(A))` denote the space of rank `r`,
+weight `k\in (q - 1)\mathbb{Z}` and type `m~(\mathrm{mod}~q-1)`
+Drinfeld modular forms, we define
+
+.. MATH::
+
+    M^{r, 0}(\mathrm{GL}_r(A)) :=
+    \bigoplus_{k\in ZZ} M_k^{r}(\mathrm{GL}_r(A))
+
+to be the graded ring of all Drinfeld modular forms of type 0. The
+graduation is given by the weight of a modular form. Similarly, we let
+`M^{r}(\mathrm{GL}_r(A)) \supset M^{r, 0}(\mathrm{GL}_r(A))` be the ring
+of all Drinfeld modular forms of arbitrary type. By
+theorem 17.5 in part III of [BRP2018]_, we have
+
+.. MATH::
+
+    M^{r, 0}(\mathrm{GL}_r(A))
+    = \mathbb{C}_{\infty}[g_1,\ldots, g_{r-1}, g_{r}].
+
+and
+
+.. MATH::
+
+    M^{r, 0}(\mathrm{GL}_r(A))
+    = \mathbb{C}_{\infty}[g_1,\ldots, g_{r-1}, h_{r}].
+
+where `h_r` is a weight `(q^r - 1)/(q - 1)` modular forms of type `1`
+which is a `(q-1)`-root of `g_r`.
+
+To create the ring of type zero forms in SageMath we use the class
+`DrinfeldModularForms`::
+
+    sage: A = GF(3)['T']
+    sage: K.<T> = Frac(A)
+    sage: M = DrinfeldModularForms(K, 3)  # rank 3
+    sage: M
+    Ring of Drinfeld modular forms of rank 3 over Fraction Field of Univariate Polynomial Ring in T over Finite Field of size 3
+
+This ring is generated by the coefficient forms that we see purely as
+symbolic objects, each of which we attach its respective weight::
+
+    sage: M.gens()
+    [g1, g2, g3]
+    sage: M.inject_variables()
+    Defining g1, g2, g3
+    sage: g1.weight()
+    2
+    sage: g2.weight()
+    8
+    sage: g3.weight()
+    26
+
+To create the ring of arbitrary types modular forms, one passes the
+keyword argument ``has_type=True``::
+
+    sage: M = DrinfeldModularForms(K, 4, has_type=True)
+    sage: M.gens()
+    [g1, g2, g3, h4]
+    sage: h4 = M.3
+    sage: h4.weight()
+    40
+
+.. RUBRIC:: Coefficients forms
+
+There are bound methods for accessing the coefficients forms at any
+`a`::
+
+    sage: A = GF(2)['T']
+    sage: K.<T> = Frac(A)
```

Comment:
The reason I did that is because the number of coefficient forms grows quickly in function of $q$, $r$ and the degree of the polynomial $a$. So to keep the output small, I chose a small $q$. I could I gone with this setting from the start, but I just don't like even prime number (no logical reason 😅)